src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+        }
+        if (anomalyDetector.getDetectionInterval() == null) {
+            missingFields.add("detector_interval");
+        }
+        if (anomalyDetector.getFeatureAttributes().isEmpty()) {
+            missingFields.add("feature_attributes");
+        }
+        if (!missingFields.isEmpty()) {
+            failuresMap.put("missing", missingFields);
+        }
+        String error = RestHandlerUtils.validateAnomalyDetector(anomalyDetector, maxAnomalyFeatures);
+        if (StringUtils.isNotBlank(error)) {
+            List<String> dupErrorsFeatures = new ArrayList<>();
+            dupErrorsFeatures.addAll(Arrays.asList(error.split("\\r?\\n")));
+            failuresMap.put("duplicates", dupErrorsFeatures);
+        }
+        if (!failuresMap.isEmpty()) {
+            validateAnomalyDetectorResponse();
+        } else {
+            validateNumberOfDetectors();
+        }
+    }
+
+    public void validateNumberOfDetectors() {
+        try {
+            QueryBuilder query = QueryBuilders.matchAllQuery();
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(query).size(0).timeout(requestTimeout);
+            SearchRequest searchRequest = new SearchRequest(ANOMALY_DETECTORS_INDEX).source(searchSourceBuilder);
+            client.search(searchRequest, ActionListener.wrap(response -> onSearchAdResponse(response), exception -> onFailure(exception)));
+        } catch (Exception e) {
+            onFailure(e);
+        }
+    }
+
+    private void onSearchAdResponse(SearchResponse response) throws IOException {
+        if (response.getHits().getTotalHits().value >= maxAnomalyDetectors) {
+            failuresMap
+                .computeIfAbsent("others", k -> new ArrayList<>())
+                .add("Can't create anomaly detector more than " + maxAnomalyDetectors);
+            validateAnomalyDetectorResponse();
+        } else {
+            searchAdInputIndices(null);
+        }
+    }
+
+    private void searchAdInputIndices(String detectorId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+            .query(QueryBuilders.matchAllQuery())
+            .size(0)
+            .timeout(requestTimeout);
+        SearchRequest searchRequest = new SearchRequest(anomalyDetector.getIndices().toArray(new String[0])).source(searchSourceBuilder);
+        client
+            .search(
+                searchRequest,
+                ActionListener
+                    .wrap(searchResponse -> onSearchAdInputIndicesResponse(searchResponse, detectorId), exception -> onFailure(exception))
+            );
+    }
+
+    private void onSearchAdInputIndicesResponse(SearchResponse response, String detectorId) throws IOException {
+        if (response.getHits().getTotalHits().value == 0) {
+            String errorMsg = "Can't create anomaly detector as no document found in indices: "
+                + Arrays.toString(anomalyDetector.getIndices().toArray(new String[0]));
```

Comment:
I think you may just use `String.format("Can't ..... in indices: %s", anomalyDetector.getIndices())`

Reply:
changed to this